ui/selectors/selectors.js

```diff
@@ -950,17 +951,17 @@ export function getDisabledRpcMethodPreferences(state) {
 }
 
 export function getShouldShowFiat(state) {
-  const isMainNet = getIsMainnet(state);
-  const isLineaMainNet = getIsLineaMainnet(state);
-  const isCustomNetwork = getIsCustomNetwork(state);
+  const currentChainId = getCurrentChainId(state);
+  const isTestnet = TEST_NETWORK_IDS.includes(currentChainId);
+  const { showFiatInTestnets } = getPreferences(state);
+
+  if (isTestnet && !showFiatInTestnets) {
```

Comment:
If we check this in the selector, can we remove the `useHideFiatForTestnet` hook entirely?